File: Assignment_4/java/proxy/src/Song.java
import java.util.Vector;

public class Song {
    private Metadata metadata;
    private String dataStream; // treated more like the name of the data file for this example

    Song(Metadata meta_data, String data_stream) {
        metadata = meta_data;
        dataStream = data_stream;
    }

    public Metadata GetMetadata() {
        return metadata;
    }

    public void SetDataStream(String data_stream) {
        dataStream = data_stream;
    }

    public void Play() {
        System.out.println("Play "  + dataStream);
    }
}

File: Assignment_4/cpp/proxy/Song.cpp
#include "Song.h"

Song::Song(std::string title, std::string album, int duration)
{
    metadata.title = title;
    metadata.album = album;
    metadata.duration = duration;
}

void Song::SetMetadata(Metadata meta_data)
{
  metadata = meta_data;
}

Metadata Song::GetMetadata()
{
  return metadata;
}
